service.py

```diff
@@ -25,10 +29,16 @@
 __version__ = __addon__.getAddonInfo('version')
 __language__ = __addon__.getLocalizedString
 
-__cwd__ = unicode(xbmc.translatePath(__addon__.getAddonInfo('path')), 'utf-8')
-__profile__ = unicode(xbmc.translatePath(__addon__.getAddonInfo('profile')), 'utf-8')
-__resource__ = unicode(xbmc.translatePath(os.path.join(__cwd__, 'resources', 'lib')), 'utf-8')
-__temp__ = unicode(xbmc.translatePath(os.path.join(__profile__, 'temp', '')), 'utf-8')
+if sys.version_info.major == 3:
+    __cwd__ = xbmc.translatePath(__addon__.getAddonInfo('path'))
+    __profile__ = xbmc.translatePath(__addon__.getAddonInfo('profile'))
+    __resource__ = xbmc.translatePath(os.path.join(__cwd__, 'resources', 'lib'))
+    __temp__ = xbmc.translatePath(os.path.join(__profile__, 'temp', ''))
+else:
+    __cwd__ = unicode(xbmc.translatePath(__addon__.getAddonInfo('path')), 'utf-8')
```

Comment:
Won't these lines also work with Python 3?

Reply:
no, the unicode function doesn't exist as all strings are unicode by default in python 3.